src/sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,412 @@
+        d_seconds = 0.5 * (n_lines - 1) * azimuth_time_interval
+        sensing_mid = sensing_start + datetime.timedelta(seconds=d_seconds)
+        az_fm_rate = get_nearest_polynomial(sensing_mid, az_fm_rate_list)
+
+        # choose nearest doppler
+        poly1d = get_nearest_polynomial(sensing_mid, doppler_list)
+        lut2d = doppler_poly1d_to_lut2d(poly1d, starting_range,
+                                        range_pxl_spacing, (n_lines, n_samples),
+                                        azimuth_time_interval)
+        doppler = Doppler(poly1d, lut2d)
+
+        # get orbit from state vector list/element tree
+        sensing_duration = datetime.timedelta(
+            seconds=n_samples * azimuth_time_interval)
+        orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
+                                osv_list)
+
+        # determine burst offset and dimensions
+        # TODO move to own method
+        first_valid_samples = [int(val) for val in burst_list_element.find('firstValidSample').text.split()]
+        last_valid_samples = [int(val) for val in burst_list_element.find('lastValidSample').text.split()]
+
+        first_valid_line = [x > 0 for x in first_valid_samples].index(True)
+        n_valid_lines = [x > 0 for x in first_valid_samples].count(True)
+        last_line = first_valid_line + n_valid_lines - 1
+
+        first_valid_sample = max(first_valid_samples[first_valid_line],
+                                 first_valid_samples[last_line])
+        last_sample = min(last_valid_samples[first_valid_line],
+                          last_valid_samples[last_line])
+
+        burst_id = f't{track_number}_{subswath_id.lower()}_b{id_burst}'
+
+        bursts[i] = Sentinel1BurstSlc(sensing_start, radar_freq, wavelength,
+                                      azimuth_steer_rate, azimuth_time_interval,
+                                      slant_range_time, starting_range,
+                                      range_sampling_rate, range_pxl_spacing,
+                                      (n_lines, n_samples), az_fm_rate, doppler,
+                                      rng_processing_bandwidth, pol, burst_id,
+                                      platform_id, center_pts[i], boundary_pts[i],
+                                      orbit, tiff_path, i, first_valid_sample,
+                                      last_sample, first_valid_line, last_line)
+
+    return bursts
+
+def burst_from_zip(zip_path: str, orbit_path: str, n_subswath: int, pol: str):
+    '''Find bursts in a Sentinel 1 zip file
+
+    Parameters:
+    -----------
+    zip_path : str
+        Path the zip file.
+    orbit_path : str
+        Path the orbit file.
+    n_subswath : int
+        Integer of subswath of desired burst. {1, 2, 3}
+    pol : str
+        Polarization of desired burst. {HH, VV, HV, VH}
+
+    Returns:
+    --------
+    bursts : list
+        List of Sentinel1BurstSlc objects found in annotation XML.
+    '''
```

Comment:
I suggest to add sanity check for n_subswath and pol right here? we know they can have very limited values. 
without those checks, the reader fails with non-intuitive messages such as this one if you ask for sub-swath 4:

```
bursts = sentinel1_reader.burst_from_zip(zip_path, orbit_path, 4, "vv")
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-20-c988331e584e> in <module>
----> 1 bursts = sentinel1_reader.burst_from_zip(zip_path, orbit_path, 4, "vv")

~/anaconda3/envs/isce3/lib/python3.9/site-packages/sentinel1_reader/sentinel1_reader.py in burst_from_zip(zip_path, orbit_path, n_subswath, pol)
    403     with zipfile.ZipFile(zip_path, 'r') as z_file:
    404         # find annotation file
--> 405         f_annotation = [f for f in z_file.namelist() if 'calibration' not in f and id_str in f and 'annotation' in f][0]
    406
    407         # find tiff file

IndexError: list index out of range
```